app/backend/src/clj/teet/file/file_tx.clj

```diff
@@ -67,11 +67,34 @@
                        :meta/modified-at :meta/modifier])
 
 (defn modify-file
-  "Modify file."
+  "Modify file. Ensures that the same task doesn't have
+  another file with the same metadata."
   [db {id :db/id :as file}]
-  (let [old-file (d/pull db modify-file-keys id)
+  (let [task-id (get-in (du/entity db id) [:task/_files 0 :db/id])
+        task-files (map first
+                        (d/q '[:find ?f
+                               :where
+                               [?t :task/files ?f]
+                               [(missing? $ ?f :meta/deleted?)]
+                               :in $ ?t]
+                             db task-id))
+        old-file (d/pull db modify-file-keys id)
         new-file (-> file
                      (select-keys modify-file-keys)
                      (update :file/part :db/id)
```

Comment:
Not part of PR, but what's going on here? Collapsing map containing db id into just the db id itself. Why?

Reply:
The map value is what we pull/select of the linked entity, but it isn't valid value for a `ref` attribute. That's why we need to just use the `:db/id`